server/src/service/collection.rs

```diff
@@ -0,0 +1,201 @@
+// org.freedesktop.Secret.Collection
+
+use std::collections::HashMap;
+
+use oo7::dbus::api::properties::Properties;
+use oo7::portal::Item;
+use serde::{Deserialize, Serialize, Serializer};
+use zbus::{dbus_interface, zvariant, Error, ObjectServer, SignalContext};
+use zvariant::{ObjectPath, OwnedObjectPath, Signature, Type};
+
+use crate::service::item;
+use crate::KEYRING;
+
+#[derive(Default, Debug, Deserialize)]
+pub struct Collection {
+    pub items: Vec<item::Item>,
+    label: String,
+    locked: bool,
+    created: u64, // how to represent date
+    modified: u64,
+    path: OwnedObjectPath,
+}
+
+#[dbus_interface(name = "org.freedesktop.Secret.Collection")]
+impl Collection {
+    pub async fn delete(&self, #[zbus(object_server)] object_server: &ObjectServer) -> ObjectPath {
+        let _ = object_server.remove::<Collection, _>(&self.path).await;
+        ObjectPath::default().into() // returning '/' until we figure out Prompt
+    }
+
+    pub async fn search_items(&self, attributes: HashMap<&str, &str>) -> Vec<Item> {
+        let items = match KEYRING.get().unwrap().search_items(&attributes).await {
+            Ok(i) => i,
+            Err(_) => todo!(),
+        };
+
+        items // currently returns Vec<oo7::portal::Item>, this should be a path
+    }
+
+    pub async fn create_item(
+        &mut self,
+        properties: Properties,
+        secret: &str,
+        replace: bool,
+    ) -> (ObjectPath, ObjectPath) {
+        let label = properties.attributes().unwrap().get("label").unwrap();
+        let mut attributes: HashMap<&str, &str> = Default::default();
+
+        for (key, value) in properties.attributes().unwrap().into_iter() {
+            attributes.insert(key.as_str(), value.as_str());
+        }
+
+        let _collection = KEYRING
```

Comment:
This doesn't return anything so instead handle the error

Reply:
Since, `create_item` doesn't return anything, what's best way to handle/match `Err` variant here? I currently have,
```
match KEYRING
            .get()
            .unwrap()
            .create_item(label, attributes.clone(), secret, replace)
            .await
        {
            Ok(_) => (), 
            Err(_) => (), // best approach to handle this error?
        }
```